### PR TITLE
feat(animation): Configurable animation motions

### PR DESCRIPTION
**Pull request by JoseLion**
This PR adds two new transition motion effects for the spotlight, making a total of 3:
- **bounce:** Slides across steps with a bouncy spring effect (default, to keep backward compatibility).
- **fade:** The spotlight fades in/out just like the tooltip component.
- **slide:** Slides across the screen with a sober linear effect.

The motion is configurable globally for the whole tour in the props of `<SpotlightTourProvider />`, or more granularly, on each step configuration.

Aside from this feature, the changes on the PR allowed us to decouple the render and animation of the spotlight and the tooltip components, improving the performance and fixing issue #77.

This PR also adds important improvements to the ESLint configuration, the codebase perse, and the tests.

### CODE REVIEW

**Review by CristhianMotoche — COMMENTED** [review]
Hey @JoseLion Overall LGTM. I just left a suggestion and a question. Let me know if they make sense.

File: src/lib/SpotlightTour.provider.tsx [hunk 1]
```diff
@@ -85,14 +94,9 @@ export const SpotlightTourProvider = React.forwardRef<SpotlightTour, SpotlightTo
...
     if (steps[index] !== undefined) {
-      const beforeResult = steps[index]?.before?.();
-      const beforePromise = isPromise(beforeResult)
-        ? beforeResult
-        : Promise.resolve();
-
       Promise.all([
-        beforePromise,
         overlay.current.hideTooltip(),
+        Promise.resolve().then(steps[index]?.before),
```
[the remark this answers is not in the document]
Comment (JoseLion) [reply src/lib/SpotlightTour.provider.tsx:99]: @CristhianMotoche good catch! This is a long-known [typescript issue](https://github.com/microsoft/TypeScript/issues/10530) where bracket notation is not discriminated correctly on type guards, so if we remove the `?` optional chaining TS complains. An easy workaround is to extract `steps[index]` to a variable and it should work 🙂

File: src/lib/SpotlightTour.provider.tsx [hunk 2]
```diff
@@ -117,7 +123,7 @@ export const SpotlightTourProvider = React.forwardRef<SpotlightTour, SpotlightTo
         ? stop()
         : renderStep(current + 1);
     }
-  }, [renderStep, current, steps.length]);
+  }, [stop, renderStep, current, steps.length]);
```
Comment (sebas1208) [inline src/lib/SpotlightTour.provider.tsx:126]: Just checking if `step` should also be added as dep?

File: src/lib/components/tour-overlay/TourOverlay.component.tsx [hunk 3]
```diff
@@ -152,38 +107,68 @@ export const TourOverlay = forwardRef<TourOverlayRef, TourOverlayProps>((props,
...
-        toValue: r,
-        useNativeDriver,
-      }),
-      Animated.timing(tipOpacity, {
-        delay: 500,
-        duration: 300,
+    tooltipRef.current?.measureInWindow((_x, _y, width, height) => {
+      setTooltipStyle(() => {
+        const align = tourStep.alignTo ?? Align.SPOT;
+        const half = (Math.max(spot.width, spot.height) / 2) * 1.15;
```
Comment (sebas1208) [inline src/lib/components/tour-overlay/TourOverlay.component.tsx:113]: What is 1.15, is it PI? Could we use the `Math` const or lets put it as a constant to know what it represents.

**Review by sebas1208 — APPROVED** [review]
LGTM!

File: src/lib/SpotlightTour.provider.tsx (hunk 2 above)
Comment (JoseLion) [reply src/lib/SpotlightTour.provider.tsx:126]: No, `step` is only in `renderStep` function scope 🙂

File: src/lib/components/tour-overlay/TourOverlay.component.tsx (hunk 3 above)
Comment (JoseLion) [reply src/lib/components/tour-overlay/TourOverlay.component.tsx:113]: The `1.15` multiplication adds 15% to the `half` const. The idea is to increase the area surrounding the element to add a gap between the spotlight and the tooltip. We do the same when we calculate the size of the spotlight, so it's a great idea to move it to a constant. Good catch! 🙂

**Pull request merged**